### DISCUSSION TITLE
Web certificate for dev/demo environment

### DISCUSSION THREAD

**Discussion by sleeman1** [wgu-opensource/osmt#138] [Asking for Help] [answered]
What's the best path for configuring a web certificate to the setup? Would you go the typical Docker/compose have the files on a volume?  Or because this is built Java/Maven is there other locations/settings we would need to change to get the app to use port 443 and a certificate?

I searched through the discussion and the help/readme files with no luck.

The larger goal is to test a dev instance using the UW's "NetID" Oauth to manage access. But before we can look at that we would want the web traffic running on port 443 using a trusted certificate,

Thanks for the help and advice!

**Comment by JohnKallies** [comment, marked as answer]
I'm hesitant to advise on best ways of configuring certificates, but I can speak with confidence to Java/Maven...

In this context, Maven builds a jar file with a Spring Boot application, and that's where Maven stops.

That Spring Boot jar file has recently been packaged in a Docker image running on CentOS 8  (https://hub.docker.com/repository/docker/wguopensource/osmt-app). The Quickstart configuration now uses that osmt-app image by default. There may be simple/typical ways to mount the right resources there, and if anything, that should help focus your search (along the lines of "https for spring boot docker image") for what best approach you decide on.

That search leads me here, which seems like an interesting resource:
* https://knasmueller.net/lets-encrypt-ssl-certificates-for-dockerized-spring-boot-in-2020

**Comment by JohnKallies** [comment]
@sleeman1 Can I please request...? If your work informs how we should package the osmt-app Docker image, would you please share it back by creating an Issue? Thanks!
